code/game/mob/living/carbon/human/human_defense.dm

```diff
@@ -628,15 +632,27 @@ bullet_act
 	else if (hit_zone == "l_leg" || hit_zone == "r_leg" || hit_zone == "r_arm" || hit_zone == "l_arm")
 		if (prob(25))
 			if (prob(60))
-				visible_message("<span class='notice'>[user] has tried to strike [src]'s [affecting.name] with [I.name] but missed!</span>")
-				return
+				if (user != src)
+					user.visible_message("<span class='danger'>[user] has tried to strike [src]'s [affecting.name] with \the [I.name] but missed!</span>", "<span class='danger'>You tried to strike [src]'s [affecting.name] with \the [I.name] but <b>missed!</span>")
+					return
+				else
+					user.visible_message("<span class = 'danger'>[user] has tried to strike \his [affecting.name] with \the [I.name] but missed!</span>", "<span class = 'danger'>You tried to strike yourself with \the [I.name] but missed!</span>")
+					return
 			else
 				affecting = get_organ("chest")
 	else if (hit_zone == "head")
 		if (prob(18))
-			visible_message("<span class='notice'>[user] has tried to strike [src]'s [affecting.name] with [I.name] but missed!</span>")
-			return
-	visible_message("<span class='danger'>[src] has been [I.attack_verb.len? pick(I.attack_verb) : "attacked"] in the [affecting.name] with [I.name] by [user]!</span>")
+			if (user != src)
+				user.visible_message("<span class='danger'>[user] has tried to strike [src]'s [affecting.name] with \the [I.name] but <b>missed!</b></span>", "<span class='danger'>You tried to strike [src]'s [affecting.name] with \the [I.name] but <b>missed!</b></span>")
+				return
+			else
+				user.visible_message("<span class = 'danger'>[user] has tried to strike \his [affecting.name] with \the [I.name] but missed!</span>", "<span class = 'danger'>You tried to strike yourself with \the [I.name] but missed!</span>")
+				return
+	if(user != src)
```

Comment:
Same here, no need to make it more complex

Reply:
No, this checks if the person is attacking themselves instead of [src], it's meant to be from different views of the clients to make it more immersive.